### PR TITLE
The script was modified to correcty assign Taskchain ACDC over merge …

### PR DESCRIPTION
**Pull request by prozober**
The script was modified to:
- Take into account the case where the request type is a resubmission, and we are dealing with a merge task.
- Assign TrustSitelists and TrustPUSitelists values.

### CODE REVIEW

File: assignProdTaskChain.py [hunk 1]
```diff
@@ -146,6 +141,8 @@ def assignRequest(url, workflow, team, site, era, procstr, procver, activity, lf
     if verbose:
         pprint(params)
 
+    #sys.exit(1)
```
Comment (amaltaro) [inline assignProdTaskChain.py]: you may remove this line :)

File: assignProdTaskChain.py [hunk 2]
```diff
@@ -294,7 +306,24 @@ def main():
...
+
+        elif schema["RequestType"] == "Resubmission":
+            # For resubmission of a merge task inside a taskchain workflow, we cannot provide the acqera and procstring
+            if "Merge" in schema["InitialTaskPath"].split("/")[-1]:
+                acqera = None
+                procstring = None
+            # For another type of task, we need to look for the acqera and procstring information inside the
+            # original workflow
+            else:
+                if any("ProcessingString" in k for k, v in schema.items()):
```
Comment (amaltaro) [inline assignProdTaskChain.py]: I'd replace it by something like `if "ProcessingString" in schema and "AcquisitionEra" in schema:`

File: assignProdTaskChain.py [hunk 3]
```diff
@@ -294,7 +306,24 @@ def main():
...
+                    procstring = schema["ProcessingString"]
+                elif not procstring:
+                    procstring = findOriginalProcessingString(url, schema["OriginalRequestName"])
+                acqera = schema["AcquisitionEra"]
```
Comment (amaltaro) [inline assignProdTaskChain.py]: then you move this line to the if just above.

File: assignProdTaskChain.py [hunk 4]
```diff
@@ -154,12 +149,24 @@ def assignRequest(url, workflow, team, site, era, procstr, procver, activity, lf
...
+    conn = httplib.HTTPSConnection(url, cert_file=os.getenv('X509_USER_PROXY'), key_file=os.getenv('X509_USER_PROXY'))
+    urn = "/reqmgr2/data/request/%s" % workflow
+    conn.request("GET", urn, headers=headers)
     r2 = conn.getresponse()
-    request = json.loads(r2.read())
-    return request
+    request = json.loads(r2.read())["result"][0]
+    return request[workflow]
+
+def findOriginalProcessingString(url,workflow):
```
Comment (amaltaro) [inline assignProdTaskChain.py]: And this method is tricky. You have to return a (or two) dictionary, like:
```
{"AcquisitionEra": {tempSchema['Task1']['TaskName']: tempSchema['Task1']['AcquisitionEra'],
                              tempSchema['Task2']['TaskName']: tempSchema['Task2']['AcquisitionEra'],
                              etc}
 "ProcessingString": {tempSchema['Task1']['TaskName']: tempSchema['Task1']['ProcessingString'],
                              tempSchema['Task2']['TaskName']: tempSchema['Task2']['ProcessingString'],
                              etc}
}
```
and then you modify you return and where you call this function according.

**Comment by vlimant** [comment]
There is something in utils.py (used by Unified) that picks AquisitionEra and ProcessingString up automatically. We can take the chance at using this instead. Where was these info taken from before ? For ACDC and clones, it should be enough to have None for Merges, and nothing specified otherwise, as it's already in the schema of the ACDC.
For assigning plain new taskchain, we would need a way to get this from the schema indeed, or in command line.

**Comment by amaltaro** [comment]
> ACDC and clones, it should be enough to have None for Merges, and nothing specified otherwise, as it's already in the schema of the ACDC.

Unless I misunderstood you, ACDC for TaskChain + Merge task must NOT specify those 2 parameters, None is a different story and will fail assignment.

**Comment by vlimant** [comment]
yes, should not specify it indeed in case of Merge. I was referring to
+                acqera = None
+                procstring = None

**Comment by amaltaro** [comment]
Paola, just to avoid any confusions. NEVER use send AcquisitionEra=None nor ProcessingString=None to reqmgr during assignment. It should either (for resubmission):
a) key/value not be present (for the merge case we fixed yesterday)
b) key/value be present with correct values (for production/processing cases)

**Comment by vlimant** [comment]
Have tested/used this for assigning properly ACDCs ?

**Pull request closed without merging**